src/Web/Slack/Experimental/Events/Types.hs

```diff
@@ -273,6 +273,41 @@ newtype MessageId = MessageId {unMessageId :: Text}
   deriving newtype (FromJSON)
   deriving stock (Show, Eq)
 
+-- | Which tab of App Home is open.
+data AppHomeTab
+  = AppHome
+  | AppHomeMessages
+  | AppHomeOther Text
```

Comment:
Should this be a newtype around `Text` with constants for `AppHomeTabHome = AppHomeTab "home"` and similar?

> One pattern I’ve seen is to use another variant that represents an unknown value. The problem with this approach is that it becomes very hard to define equality and comparisons on these types. Are `Architecture::I386` and `Architecture::Unknown("i386")` the same, or are they different? In practice, this becomes tricky very quickly.

https://sunshowers.io/posts/open-closed-universes/